### PR TITLE
446 encounter listview

### LINKED ISSUE #446: Encounter Tab / List View
Add encounter tab that leads to an encounter list view

Image

### PR DESCRIPTION
**Pull request by clemens-msupply**
- updates encounter schemas
- improves / updates encounter api: pagination, sort 
- encounter tab with list view
- encounter detail modal

There is an issue with shared sort option across tabs:
[#469](https://github.com/openmsupply/open-msupply/issues/469)

Closes #446

### CODE REVIEW

**Comment by mark-prins** [comment]
I noticed this jank when loading the modals

patient-jank

and have put in a suggestion: abb3aeb62
Too many skeletons 😉  The skeleton is just for the loading of the detail view - no need to show it for the others too

File: client/packages/system/src/Patient/DetailView/DetailView.tsx [hunk 1]
```diff
@@ -119,6 +122,11 @@ export const PatientDetailView: FC = () => {
                 label={t('label.programs')}
                 tabIndex={-1}
               />
+              <Tab
+                value={Tabs.Encounters}
+                label={t('label.encounters')}
+                tabIndex={-1}
```
Comment (mark-prins) [inline client/packages/system/src/Patient/DetailView/DetailView.tsx:128]: shouldn't have the negative tab index - that's just for the default. It's an accident that it is on the programs one.. sorry!

File: client/packages/system/src/Patient/Encounter/ListView.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  ColumnAlign,
+} from '@openmsupply-client/common';
+import { EncounterFragment, useEncounter } from './api';
+import { usePatientModalStore } from '../hooks';
+import { PatientModal } from '../DetailView';
+
+type EncounterFragmentWithId = { id: string } & EncounterFragment;
+
+const EncounterListComponent: FC = () => {
+  const {
```
Comment (mark-prins) [inline client/packages/system/src/Patient/Encounter/ListView.tsx:20]: this is where the combination of tabs and url query params falls down.
If you go to encounters and sort by something, then go to the programs tab; you get an error shown: `Error: Invalid value for argument "sort.key", enumeration type "ProgramSortFieldInput" does not contain the value "program"
`

the url params will need to be reset to allowable values on the changing to a different tab, or we remove url support for the sorting of the list views which are in tabs

File: server/repository/migrations/postgres/2022-06-07T17-00_create_encounter_table/up.sql [hunk 3]
```diff
@@ -1,16 +1,16 @@
 CREATE TYPE encounter_status AS ENUM (
     'SCHEDULED',
-    'ONGOING',
-    'FINISHED',
-    'CANCELED',
-    'MISSED'
+    'DONE',
+    'CANCELED'
```
Comment (mark-prins) [inline server/repository/migrations/postgres/2022-06-07T17-00_create_encounter_table/up.sql:4]: :nit
```suggestion
    'CANCELLED'
```

**Review by mark-prins — APPROVED** [review]
🎉  looking really good!

A couple of things - though I think we can merge and tidy them later. The most obvious is the UI when opening the modal and I've put in a fix for that - next is the url query params; maybe a new issue?

and the usual nit-picky English spelling thing 😉

File: client/packages/system/src/Patient/Encounter/ListView.tsx (hunk 2 above)
Comment (clemens-msupply) [reply client/packages/system/src/Patient/Encounter/ListView.tsx:20]: yes noticed it as well I created an issue for it: https://github.com/openmsupply/open-msupply/issues/469

File: client/packages/system/src/Patient/DetailView/DetailView.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/system/src/Patient/DetailView/DetailView.tsx:128]: done

File: server/repository/migrations/postgres/2022-06-07T17-00_create_encounter_table/up.sql (hunk 3 above)
Comment (clemens-msupply) [reply server/repository/migrations/postgres/2022-06-07T17-00_create_encounter_table/up.sql]: done

**Comment by clemens-msupply** [comment]
Thanks I made the change/ created an issue for the url problem

**Pull request merged**